UInt32Value is not supported in generated unit tests

It seems like there are issues in properly handliing Value classes by proto_plus.

Affected APIs:
```
//google/cloud/bigquery/connection/v1beta1/connection_py_gapic_test
//google/cloud/bigquery/v2/bigquery_py_gapic_test
```

Failure output:

```
google/cloud/bigquery/connection/v1beta1/connection_py_gapic_srcjar.py/tests/unit/gapic/connection_v1beta1/test_connection_service.py:1231: AssertionError
____________________ test_list_connections_flattened_async _____________________

    @pytest.mark.asyncio
    async def test_list_connections_flattened_async():
        client = ConnectionServiceAsyncClient(
            credentials=ga_credentials.AnonymousCredentials(),
        )
    
        # Mock the actual call within the gRPC stub, and fake the request.
        with mock.patch.object(
                type(client.transport.list_connections),
                '__call__') as call:
            # Designate an appropriate return value for the call.
            call.return_value = connection.ListConnectionsResponse()
    
            call.return_value = grpc_helpers_async.FakeUnaryUnaryCall(connection.ListConnectionsResponse())
            # Call the method with a truthy value for each flattened field,
            # using the keyword arguments to the method.
            response = await client.list_connections(
                parent='parent_value',
                max_results=wrappers_pb2.UInt32Value(value=541),
            )
    
            # Establish that the underlying call was made with the expected
            # request object values.
            assert len(call.mock_calls)
            _, args, _ = call.mock_calls[0]
            arg = args[0].parent
            mock_val = 'parent_value'
            assert arg == mock_val
            arg = args[0].max_results
            mock_val = wrappers_pb2.UInt32Value(value=541)
>           assert arg == mock_val
E           assert 541 == value: 541\n

google/cloud/bigquery/connection/v1beta1/connection_py_gapic_srcjar.py/tests/unit/gapic/connection_v1beta1/test_connection_service.py:1278: AssertionError
=========================== short test summary info ============================
FAILED google/cloud/bigquery/connection/v1beta1/connection_py_gapic_srcjar.py/tests/unit/gapic/connection_v1beta1/test_connection_service.py::test_list_connections_flattened
FAILED google/cloud/bigquery/connection/v1beta1/connection_py_gapic_srcjar.py/tests/unit/gapic/connection_v1beta1/test_connection_service.py::test_list_connections_flattened_async
2 failed, 171 passed, 39 warnings in 1.15s
```

--- a/.bazelrc
+++ b/.bazelrc
@@ -1,0 +1,2 @@
+# New boringssl requires C++14
+build --repo_env=BAZEL_CXXOPTS="-std=c++14"

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -50,6 +50,20 @@ load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
 
 protobuf_deps()
 
+# Import boringssl explicitly to override what gRPC imports as its dependency.
+# Boringssl build fails on gcc12 without this fix:
+# https://github.com/google/boringssl/commit/8462a367bb57e9524c3d8eca9c62733c63a63cf4,
+# which is present only in the newest version of boringssl, not the one imported
+# by gRPC. Remove this import once gRPC depends on a newer version.
+http_archive(
+    name = "boringssl",
+    sha256 = "b460f8673f3393e58ce506e9cdde7f2c3b2575b075f214cb819fb57d809f052b",
+    strip_prefix = "boringssl-bb41bc007079982da419c0ec3186e510cbcf09d0",
+    urls = [
+        "https://github.com/google/boringssl/archive/bb41bc007079982da419c0ec3186e510cbcf09d0.zip",
+    ],
+)
+
 #
 # Import grpc as a native bazel dependency. This avoids duplication and also
 # speeds up loading phase a lot (otherwise python_rules will be building grpcio

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2
@@ -21,7 +21,7 @@ import json
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 {% if 'rest' in opts.transport %}
 from requests import Response
 from requests import Request, PreparedRequest

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
@@ -400,6 +400,8 @@ def test_{{ method_name }}_flattened():
         assert TimestampRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
         {% elif field.ident|string() == 'duration_pb2.Duration' %}
         assert DurationRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
+        {% elif (field.ident|string()).startswith('wrappers_pb2.') %}
+        assert wrappers.{{ (field.ident|string())[13:] }}Rule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
         {% else %}
         arg = args[0].{{ key }}
         mock_val = {{ field.mock_value }}
@@ -493,6 +495,8 @@ async def test_{{ method_name }}_flattened_async():
         assert TimestampRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
         {% elif field.ident|string() == 'duration_pb2.Duration' %}
         assert DurationRule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
+        {% elif (field.ident|string()).startswith('wrappers_pb2.') %}
+        assert wrappers.{{ (field.ident|string())[13:] }}Rule().to_proto(args[0].{{ key }}) == {{ field.mock_value }}
         {% else %}
         arg = args[0].{{ key }}
         mock_val = {{ field.mock_value }}

--- a/tests/integration/goldens/asset/tests/unit/gapic/asset_v1/test_asset_service.py
+++ b/tests/integration/goldens/asset/tests/unit/gapic/asset_v1/test_asset_service.py
@@ -26,7 +26,7 @@ from grpc.experimental import aio
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 
 from google.api_core import client_options
 from google.api_core import exceptions as core_exceptions

--- a/tests/integration/goldens/credentials/tests/unit/gapic/credentials_v1/test_iam_credentials.py
+++ b/tests/integration/goldens/credentials/tests/unit/gapic/credentials_v1/test_iam_credentials.py
@@ -26,7 +26,7 @@ from grpc.experimental import aio
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 
 from google.api_core import client_options
 from google.api_core import exceptions as core_exceptions

--- a/tests/integration/goldens/eventarc/tests/unit/gapic/eventarc_v1/test_eventarc.py
+++ b/tests/integration/goldens/eventarc/tests/unit/gapic/eventarc_v1/test_eventarc.py
@@ -26,7 +26,7 @@ from grpc.experimental import aio
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 
 from google.api_core import client_options
 from google.api_core import exceptions as core_exceptions

--- a/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_config_service_v2.py
+++ b/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_config_service_v2.py
@@ -26,7 +26,7 @@ from grpc.experimental import aio
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 
 from google.api_core import client_options
 from google.api_core import exceptions as core_exceptions

--- a/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_logging_service_v2.py
+++ b/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_logging_service_v2.py
@@ -26,7 +26,7 @@ from grpc.experimental import aio
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 
 from google.api import monitored_resource_pb2  # type: ignore
 from google.api_core import client_options

--- a/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_metrics_service_v2.py
+++ b/tests/integration/goldens/logging/tests/unit/gapic/logging_v2/test_metrics_service_v2.py
@@ -26,7 +26,7 @@ from grpc.experimental import aio
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 
 from google.api import distribution_pb2  # type: ignore
 from google.api import label_pb2  # type: ignore

--- a/tests/integration/goldens/redis/tests/unit/gapic/redis_v1/test_cloud_redis.py
+++ b/tests/integration/goldens/redis/tests/unit/gapic/redis_v1/test_cloud_redis.py
@@ -26,7 +26,7 @@ from grpc.experimental import aio
 import math
 import pytest
 from proto.marshal.rules.dates import DurationRule, TimestampRule
-
+from proto.marshal.rules import wrappers
 
 from google.api_core import client_options
 from google.api_core import exceptions as core_exceptions

